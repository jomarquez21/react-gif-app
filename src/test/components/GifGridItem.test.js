import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from './../../components/GifGridItem';
import toJson from 'enzyme-to-json';

describe('Test funtional on `GifGridItem`', () => {
  const title = 'JoMarquez';
  const url = 'google.com';
  const component = shallow(<GifGridItem title={title} url={url} />);

  test('Ensure that GifGridItem is correctly rendered', () => {
    expect(true).toBe(true);

    expect(toJson(component)).toMatchSnapshot();
  });

  test('Should render a paragraph', () => {
    const p = component.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('Deberia tener la imagen igual a la url y alt de los props', () => {
    const img = component.find('img');

    const imgProps = img.props();

    expect(imgProps.src).toBe(url);
    expect(imgProps.alt).toBe(title);
  });

  test('Debe tener animate__fadeIn', () => {
    const div = component.find('div');

    const divProps = div.props();
    expect(divProps.className.includes('animate__fadeIn')).toBe(true);
  });
});