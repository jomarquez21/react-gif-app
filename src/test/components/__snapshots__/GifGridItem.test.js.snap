// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test funtional on \`GifGridItem\` Ensure that GifGridItem is correctly rendered 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="JoMarquez"
    src="google.com"
  />
  <p>
     
    JoMarquez
     
  </p>
</div>
`;
